lab-doug/app/component/main/index.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+require('./main.scss');
+
+const angular = require('angular');
+const widgetApp = angular.module('widgetApp')
+widgetApp.directive('appMain', function(){
+  return {
+    restrict: 'E',
+    replace: true,
+    template: require('./main.html'),
+    controller: 'MainController',
+    controllerAs: 'mainCtrl',
+    bindToController: true,
+    scope: {}
+  };
+});
+
+widgetApp.controller('MainController', ['listService', MainController]);
+/**
+ * MainController - populates the main.html template
+ *
+ * @param  {object} "listService" provides ability to perform CRUD on lists in the backend db as well as updating listService "list" array. The "listService" object is passed by reference as this parameter.
+ * @return {array} creates the "lists" property on the MainController which references the "lists" array in listService object. If the promise "resolve" sucessfully, then a populated "lists" array is returned.   The main.html template uses angular data binding to this MainController.lists property, to populate current lists on the page.
+ */
+function MainController(listService){
+
```

Comment:
If you're going to attach properties to MainController later, I would include them at the beginning as well for clarity and to avoid bugs. For example, I would have this.lists = [] at the beginning and then reset it's value to the list you get from the server when you get the server's response.